docs/site/site-proposal.md

```diff
@@ -0,0 +1,159 @@
+  [Composition and responsibility of LoopBack 4 constructs for a typical application](https://github.com/strongloop/loopback-next/pull/2925)
+  in the original proposal.
+
+  _need a diagram that shows the relationship among all the building blocks. it
+  shows:_
+
+  - logical composition of the building blocks (diagram 1)
+  - how the request flow through different tiers (diagram 2)
+
+  Subitems include:
+
+  - [Application](Application.md)
+  - [Server](Server.md)
+  - [Controller](Controllers.md)
+  - Model & Model Relation: This page has a combination of [Model.md](Model.md)
+    and [Relations.md](Relations.md)
+    - [HasMany Relation](HasMany-relation.md)
+    - [BelongsTo Relation](BelongsTo-relation.md)
+    - [HasOne Relation](hasOne-relation.md)
+  - [Repository](Repositories.md)
+  - Service Proxy
+  - Datasource & Connectors
+  - Observers & Interceptors
+
+**Application life cycles**
+
+- Create: includes scaffolding
+  - download the cli module
+  - create project skeleton
+  - scaffold application, create model, add business logics in controller.
+- Add more artifacts
+  - start to add building blocks
+- Build
+  - part of `npm start`
+  - build step will transpile everything to javascript
+- Boot
+  - `app.boot()`
+  - load the artifact by convention from well know folders and populate into the
+    application context
+- Start
+  - `app.start()`
+  - LB application does the start event
+  - activation of endpoints
+  - start to listen on the transport and start to accept requests
+  - lifecycle event, e.g. start a timer to clean up cache periodically
+- Request/Response processing
+  - after start, requests can come in.
+  - can start accept request and produce response
+  - raymond has a diagram from server > sequence of actions> controller >
+    datasource > connector
+- Stop
+  - deactivate the endpoints. shutting down the application
+
+**Developer Experiences**
+
+_Development and Deployment stories_
+
+_show how to add more integration capabilities to your app_
+
+- Create REST APIs
+  - Top-down vs Bottom up
+  - OpenAPI
+- Access databases
+- Call other services
+- Integrate with infrastructure
+  - Authentication
+  - Authorization
+  - Caching
+  - Distributed tracing
+- Debugging
+- Deployment
+
+**Behind the scene plumbing**
+
+_what's the plumbing that we get all the artifacts working together_
+
+- Context
+- Binding
+- Dependency injection
+- Component
+
+**Request/response processing flow**
+
+_Programming model guide_
+
+It serves 2 purposes:
+
+1. for extension devloper on how to extend LB
+2. for application developer to understand the processing flow
+
+- Transport -> Server -> Sequence of actions -> Controller -> Repository ->
+  DataSource -> Connector -> Database
+- Express middleware
+- Sequence of actions
+  - Routing
+  - Parsing
+  - Invoking
+  - Sending
+  - Rejecting
+- Interceptors
+
+**Extending LoopBack**
+
+- Extension point/extension
+- Discovering and ordering
+- Chain of handling
+
```

Comment:
At the moment, we have the following sections in `Extending LoopBack`:

<img width="204" alt="Screen Shot 2019-06-24 at 09 27 26" src="https://user-images.githubusercontent.com/1140553/59999459-5182d180-9662-11e9-84dc-a29890a08521.png">

What's the plan - are we going to preserve them unchanged?

From my point of view, I feel it's important to provide good documentation for extension developers, showing them best practices for building npm modules providing reusable functionality for LB4 applications.

Reply:
I've added them under "Extending LoopBack". Thanks.